test/test_csvs.py

```diff
@@ -0,0 +1,291 @@
+    # ###########################################
+
+    # the PSVs are built from the CSVs in setuptest.sh to make them pipe-separated files
+    # due to the "local" file using commas inside the fields of a handful of records
+
+    def test_globalFileExists(self):
+        self.assertTrue(os.path.exists(self._getpath('data_qc_global_range_values.psv')))
+
+    def test_gradientFileExists(self):
+        self.assertTrue(os.path.exists(self._getpath('data_qc_gradient_test_values.psv')))
+
+    def test_localFileExists(self):
+        self.assertTrue(os.path.exists(self._getpath('data_qc_local_range_values.psv')))
+
+    def test_spikeFileExists(self):
+        self.assertTrue(os.path.exists(self._getpath('data_qc_spike_test_values.psv')))
+
+    def test_stuckFileExists(self):
+        self.assertTrue(os.path.exists(self._getpath('data_qc_stuck_test_values.psv')))
+
+    def test_trendFileExists(self):
+        self.assertTrue(os.path.exists(self._getpath('data_qc_trend_test_values.psv')))
+
+    # these files are generated within setuptest.sh from the asset-management
+    # and preload-database repositories
+
+    def test_parameterFileExistsAndIsPopulated(self):
+        self.assertTrue(os.path.exists(self._getpath('parameter.txt')))
+        parameters = self._get_parameters()
+        self.assertTrue(len(parameters) > 0)
+
+    def test_refdesFileExistsAndIsPopulated(self):
+        self.assertTrue(os.path.exists(self._getpath('refdes.txt')))
+        refdes = self._get_ref_designators()
+        self.assertTrue(len(refdes) > 0)
+
+    def test_typeFileExistsAndIsPopulated(self):
+        self.assertTrue(os.path.exists(self._getpath('type.txt')))
+        types = self._get_types()
+        self.assertTrue(len(types) > 0)
+
+    def test_unitFileExistsAndIsPopulated(self):
+        self.assertTrue(os.path.exists(self._getpath('unit.txt')))
+        units = self._get_units()
+        self.assertTrue(len(units) > 0)
+
+    def test_validateGlobalRangeFile(self):
+        validfile = True # file is valid until proven otherwise
+        header = None # obtained from 1st record
+        # the numbers of the fields in each data record needing validation
+        flds_to_validate = None # established from 1st record
+
+        with io.open(self._getpath('data_qc_global_range_values.psv'),mode='rt',encoding='iso-8859-1') as f:
+            for index,l in enumerate(f):
+                line = l.strip() # drop the trailing newline
+                # the 1st record contains the headers
+                if header is None:
+                    header = line.split('|')
+                    flds_to_validate = range(len(header)) # validate all fields
+                    continue
+                invalidlist = self._validate_line(header, flds_to_validate, line)
+                if invalidlist:
+                    validfile = False
+                    print('Global record invalid ' + str(index+1) + ': ' + line)
```

Comment:
We should be in the mind-set of doing nosetests here, not debugging. Are the `print` lines necessary? If so, they should be incorporated into the `assertTrue` optional message argument:

`self.assertTrue(valid_file, 'bad stuff')`